src/pages/profile.jsx

```diff
@@ -1,56 +1,250 @@
 import React from 'react';
-import Framework7 from 'framework7/framework7.esm.bundle.js';
+
+//import styling
+import '../css/profile.css';
+import '../css/toolbar.css';
+
+//import components
+import Button from '../components/shared/Button/StyledButton';
+import ProfilePageHeader from '../components/ProfilePage/ProfilePageHeader/ProfilePageHeader';
+import ProfileInformation from '../components/shared/InformationBox/InformationBox';
+import ProfileSkills from '../components/ProfilePage/ProfileSkills/ProfileSkills';
+import Header from '../components/Header/Header.jsx';
+import AboutForm from '../components/ProfilePage/ProfilePageAboutForm/ProfilePageAboutForm';
+import BioForm from '../components/ProfilePage/ProfilePageBioForm/ProfilePageBioForm';
+import ProfilePageSkillsForm from '../components/ProfilePage/ProfilePageSkillsForm/ProfilePageSkillsForm';
+import EditableButton from '../components/ProfilePage/ProfilePageEditableButton/ProfilePageEditableButton';
+
 import {
     Page,
-    Navbar,
-    Block,
-    BlockTitle,
-    List,
-    ListItem,
+    Row,
     Toolbar,
-    Link
+    Link,
+    Popup,
+    Navbar,
+    NavRight,
+    NavLeft
 } from 'framework7-react';
 
-import Header from '../components/Header/Header.jsx';
-
-import '../css/toolbar.css';
-
 class Profile extends React.Component {
     constructor() {
         super();
         this.state = {
-            user: {}
+            profileDescription:
+                'Jeg er tidligere finalist i Norsk Informatikkolympiade, tar mastergrad i datateknologi ved NTNU, og har hovedansvaret for utvikling av InMarkets nettsider og app.',
+            skills: [
+                { text: 'Hacking', rating: 4.1 },
+                { text: 'Programmering', rating: 5.0 },
+                { text: 'Lederegenskaper', rating: 4.3 }
+            ],
+            activeSkills: ['Hacking', 'Programmering'],
+            rating: 4.5,
+            popupOpen: false,
+            currentForm: '',
+            firstName: 'Ask',
+            lastName: 'Kolltveit',
+            birthDate: '1997-01-01',
+            role: 'Teknologidirektør',
+            institution: 'Inmarket AS',
+            formerEmployers: ''
         };
     }
 
     componentDidMount() {
-        fetch(`http://localhost/api/users/${this.$f7route.params.id}`)
+        let id = this.$f7route.params.id;
+        if (id === 'me' || id === '') {
+            id = JSON.parse(atob(localStorage.jwt.split('.')[1])).sub;
+        }
+
+        const url = gConfig.url + '/users/' + id;
+        console.log(url);
+
+        fetch(url, { headers: { authorization: localStorage.jwt } })
             .then(res => {
                 return res.json();
             })
             .then(user => {
-                this.setState({ user: user.data });
+                this.setInitialData(user.data);
+                console.log(user.data);
             });
     }
 
+    setInitialData(data) {
```

Comment:
Unødvendig med denne metoden når det eneste den gjør er å kjøre `this.setState()`?